pkg/apis/resource/types.go

```diff
@@ -584,6 +664,10 @@ type DeviceConstraint struct {
 	// constraint. If this is not specified, this constraint applies to all
 	// requests in this claim.
 	//
+	// References to subrequests must include the name of both the main request
+	// and the subrequest using the format <main request>/<subrequest>. If just
+	// the main request is given, the constraint applies to all subrequests.
```

Comment:
At first glance, the "must include the name of both the main request and the subrequest" seems to contradict the "just the main request is given" - it is possible to reference subrequests without the subrequest name?!

Perhaps rephrase:

```
// References to subrequests must include the name of the main request
// and may include the subrequest using the format <main request>[/<subrequest>]. If just
// the main request is given, the constraint applies to all subrequests.
```

Same below.

Reply:
Yeah, agree that the suggest language is clearer. I've updated all places where this language was used in both the internal types and the v1beta1 and v1alpha3 APIs.